src/utilcode/prettyprintsig.cpp

```diff
@@ -733,7 +729,14 @@ static HRESULT PrettyPrintTypeA(
             IfFailGo(appendStrA(out, pN));
         }
         break;
-        
+
+    case ELEMENT_TYPE_CMOD_REQD:
+        str = "required_modifier ";
+        IfFailGo(appendStrA(out, str));
+        typePtr += CorSigUncompressToken(typePtr, &tk);
```

Comment:
The type token should be printed not skipped. It can be something like:

```
str = "required_modifier ";
IfFailGo(appendStrA(out, str));
IfFailGo(PrettyPrintTypeA(typePtr, (typeEnd - typePtr), out, pIMDI));
goto AGAIN;
```

And add `AGAIN` label right before `elt = *typePtr++;` above.

Reply:
I couldn't get this to work by jumping back to an AGAIN label, it always prints garbage. I does work however if I copy the contents of the DO_CLASS label after the required_modifier and then add a PrettyPrintTypeA, see the second changeset.

How can this be simplified, should I just extract this into a seperate method?

Reply:
Yes, extracting it into a separate method would definitely look better.

Reply:
Done.